pkg/process/manager.go

```diff
@@ -603,12 +603,34 @@ func (m *Manager) Reconnect(ctx context.Context) ([]*daemon.Daemon, error) {
 		m.daemonStates.RecoverDaemonState(d)
 
 		d.Once = &sync.Once{}
+
+		defer func() {
+			// `CheckStatus->ensureClient` only checks if socket file is existed when building http client.
+			// But the socket file may be residual and will be cleared before starting a new nydusd.
+			// So clear the client by assigning nil
+			d.Client = nil
+			d.Once = &sync.Once{}
```

Comment:
Ditto.

Reply:
Because during iterating daemons in DB, their unix sockets might be deleted before restarting a new nydusd. So HTTP client has to be reinitialized

Reply:
Okay, can we put it into something like `d.DestroyClient`? Ditto.